### PR TITLE
Add rsnapshot package

### PR DESCRIPTION
**Pull request by peeter123**
_Motivation:_ Added [rsnapshot](http://rsnapshot.org) for incremental backups. 

Rsnapshot is a filesystem snapshot utility based on rsync. rsnapshot makes it easy to make periodic snapshots of local machines, and remote machines over ssh. The code makes extensive use of hard links whenever possible, to greatly reduce the disk space required.
### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

File: cross/rsnapshot/Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+PKG_DIST_NAME = $(PKG_DIR).$(PKG_EXT)
+PKG_DIST_SITE = http://rsnapshot.org/downloads
+
+HOMEPAGE = http://rsnapshot.org/
+COMMENT  = Rsnapshot is a filesystem snapshot utility. It can take incremental snapshots of local and remote filesystems. 
+LICENSE  = GNU-GPL
+
+GNU_CONFIGURE = 1
+CONFIGURE_ARGS = --prefix=$(INSTALL_PREFIX) --with-perl=/usr/bin/perl --with-rsync=/usr/bin/rsync --with-ssh=/usr/bin/ssh
```
Comment (Dr-Bean) [inline cross/rsnapshot/Makefile]: `--prefix=$(INSTALL_PREFIX)` should already be set via configure.mk. See https://github.com/SynoCommunity/spksrc/blob/master/mk/spksrc.configure.mk#L40. In other words, it should work without explicitly defining the prefix here.

File: cross/rsnapshot/PLIST [hunk 2]
```diff
@@ -0,0 +1,3 @@
+rsc:bin/rsnapshot
```
Comment (Dr-Bean) [inline cross/rsnapshot/PLIST]: Use `bin:` as prefix for binary files.

File: cross/rsnapshot/PLIST [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+rsc:bin/rsnapshot-diff
```
Comment (Dr-Bean) [inline cross/rsnapshot/PLIST]: Use `bin:` as prefix for binary files.

File: spk/rsnapshot/Makefile [hunk 4]
```diff
@@ -0,0 +1,24 @@
+SPK_NAME = rsnapshot
+SPK_VERS = 1.4.2
+SPK_REV = 3
```
Comment (Dr-Bean) [inline spk/rsnapshot/Makefile]: `1`

File: spk/rsnapshot/Makefile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+SPK_ICON = src/rdiff-backup.png
+
+DEPENDS = cross/$(SPK_NAME)
+SPK_DEPENDS = "Perl"
+
+MAINTAINER = SynoCommunity
```
Comment (Dr-Bean) [inline spk/rsnapshot/Makefile]: You're the package maintainer.

File: spk/rsnapshot/src/installer.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    mkdir -p ${TMP_DIR}
+    cp ${INSTALL_DIR}/etc/rsnapshot.conf ${TMP_DIR}
+
+    exit 0
+}
+
+postupgrade ()
+{
+    # Restore the config file
+    cp ${TMP_DIR}/rsnapshot.conf ${INSTALL_DIR}/etc/rsnapshot.conf
```
Comment (Dr-Bean) [inline spk/rsnapshot/src/installer.sh]: L27 is executed before this line during a package upgrade, so the file will already exist. You have to make sure the copy will never fail.

File: spk/rsnapshot/src/installer.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    rm -rf ${TMP_DIR}
+
+    # Upgrade config file to new version
+    /usr/local/bin/rsnapshot upgrade-config-file
```
Comment (Dr-Bean) [inline spk/rsnapshot/src/installer.sh]: Use `${INSTALL_DIR}/bin/rsnapshot`, not the symlink.

File: spk/rsnapshot/src/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,2 @@
+# Cross compiled packages dependencies
```
Comment (Dr-Bean) [inline spk/rsnapshot/src/requirements.txt]: This file is only required when cross-compiling Python dependencies. From the PR, I gather that this does not apply ;)

File: spk/rsnapshot/src/installer.sh [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    exit 0
+}
+
+postuninst ()
+{
+    # Remove link
+    rm -f ${INSTALL_DIR}
+    rm /usr/local/bin/rsnapshot
+    rm /usr/local/bin/rsnapshot-diff
+    exit 0
```
Comment (Dr-Bean) [inline spk/rsnapshot/src/installer.sh]: Please add an empty line here.

File: spk/rsnapshot/src/installer.sh [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+postupgrade ()
+{
+    # Restore the config file
+    cp ${TMP_DIR}/rsnapshot.conf ${INSTALL_DIR}/etc/rsnapshot.conf
+    rm -rf ${TMP_DIR}
+
+    # Upgrade config file to new version
+    /usr/local/bin/rsnapshot upgrade-config-file
+    exit 0
```
Comment (Dr-Bean) [inline spk/rsnapshot/src/installer.sh]: One empty line in between here.

**Comment by peeter123** [comment]
Thanks for your comments! I will update the PR sometime this week.

**Comment by Dr-Bean** [comment]
Haven't verified that it works, but the code looks good to me :)

**Comment by Dr-Bean** [comment]
Merged with 0b11b2413cd10d4cdf29225ef3d8f6c47ccfd6f9. Thanks :) Package will be published in a bit.

**Pull request merged**